tests/integration/cli/commands/test_issue_1688.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+import subprocess
+
+from pex.cli.testing import run_pex3
+from pex.interpreter import PythonInterpreter
+from pex.pex_info import PexInfo
+from pex.resolve import lockfile
+from pex.testing import run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+def test_multiplatform_sdist(
+    tmpdir,  # type: Any
+    py27,  # type: PythonInterpreter
+    py37,  # type: PythonInterpreter
+    py310,  # type: PythonInterpreter
+):
+    # type: (...) -> None
+
+    all_interpreters = (py27, py37, py310)
+    python_path = ":".join((interp.binary for interp in all_interpreters))
+    interpreter_selection_args = [
+        "--python-path",
+        python_path,
+        "--interpreter-constraint",
+        ">=2.7,<3.11",
+    ]
+
+    lock = os.path.join(str(tmpdir), "lock")
+    run_pex3(
+        "lock",
+        "create",
+        "--style",
+        "universal",
+        "--no-wheel",
+        "psutil==5.9.0",
```

Comment:
I usually like to use the exact issue case in the test, but `thrift` won't compile on machines without it; so I picked `psutil` as the stand in.

That said, the issue case now works:
```console
$ python -mpex.cli lock create thrift==0.15.0 --style universal --interpreter-constraint ">=3.8" --indent 2 -o 1688.lock.json
$ python -mpex --interpreter-constraint ">=3.8" --lock 1688.lock.json -o 1688.pex
$ pex-tools 1688.pex info -i2 | jq .distributions
{
  "six-1.16.0-py2.py3-none-any.whl": "3e1c439c88d2e7681372427bab751b3fc99969891e95a714fed9604bf7710213",
  "thrift-0.15.0-cp310-cp310-linux_x86_64.whl": "02c33676139b3a57a70fae0f4047f70a3c70041e2ea582fcea8c329e78a6ce0b",
  "thrift-0.15.0-cp38-cp38-linux_x86_64.whl": "a96eeeb6a5fb4b09dd57dc5d58b8ce116178aa1e108008c1b8a69daa62e0c003",
  "thrift-0.15.0-cp39-cp39-linux_x86_64.whl": "62ba88f27298dee8608178c562db89b4240564ee2bf059f2be3256283f07bcf9"
}
$ for v in 8 9 10; do python3.$v 1688.pex -c 'import thrift; print(thrift.__file__)'; done
/home/jsirois/.pex/installed_wheels/a983a6fc6296b5886a2b5101f3ff6214fd24ec3a8be4ca17c3f03576d832f77b/thrift-0.15.0-cp38-cp38-linux_x86_64.whl/thrift/__init__.py
/home/jsirois/.pex/installed_wheels/ea4a5d722f45e2d779bd21ac00547259ef6e412b146280f267bf4855115624ba/thrift-0.15.0-cp39-cp39-linux_x86_64.whl/thrift/__init__.py
/home/jsirois/.pex/installed_wheels/46362f8f4ffd762859bd30595338c6b43783075942ed204cdedab84787309a57/thrift-0.15.0-cp310-cp310-linux_x86_64.whl/thrift/__init__.py
```